src/main/java/com/nutomic/syncthingandroid/activities/DeviceActivity.java

```diff
@@ -207,6 +250,24 @@ public void onPause() {
     public void onSaveInstanceState(Bundle outState) {
         super.onSaveInstanceState(outState);
         outState.putString("device", new Gson().toJson(mDevice));
+        if (mIsCreateMode){
+            outState.putBoolean(IS_SHOWING_DISCARD_DIALOG, mIsShowingDiscardDialog);
+            if(mDiscardDialog != null){
+                mDiscardDialog.cancel();
+            }
```

Comment:
If the dialog is not canceled, then I get an error saying that a window was leaked. But the app doesn't crash.